include/seqan3/alignment/pairwise/policy/find_optimum_policy.hpp

```diff
@@ -77,103 +69,118 @@ class find_optimum_policy
     //!\}
 
 protected:
-    /*!\brief Checks every cell of the dynamic programming matrix.
+    /*!\brief Checks if a given cell is a new optimum in the alignment.
+     * \tparam cell_t The type of the cell.
      * \tparam score_t The type of the score.
-     * \param[in] current_cell The current cell.
-     * \param[in,out] cache The cache with the current optimum to update.
+     *
+     * \param[in] current_cell The current computed cell.
+     * \param[in,out] state The state with the current optimum to update.
      *
      * \details
      *
      * This function resolves to a "NO-OP" function if the trait for searching every cell is set to std::false_type.
      */
     template <typename cell_t, typename score_t>
-    constexpr void check_score([[maybe_unused]] cell_t const & current_cell,
-                               [[maybe_unused]] alignment_algorithm_cache<score_t> & cache) const noexcept
+    constexpr void check_score_of_cell([[maybe_unused]] cell_t const & current_cell,
+                               [[maybe_unused]] alignment_algorithm_state<score_t> & state) const noexcept
```

Comment:
:warning: alignment expert needed to fix this :warning:  :man_astronaut: rene is incomming